viewing_party/party.py

```diff
@@ -1,23 +1,171 @@
 # ------------- WAVE 1 --------------------
+"""Create and modify movie data"""
+
 
 def create_movie(title, genre, rating):
-    pass
+    """create a movie dictionary"""
+    if title is None or genre is None or rating is None:
+        return None
+    return {"title": title, "genre": genre, "rating": rating}
+
+
+def add_to_watched(user_data, movie):
+    """append movie dictionary to user_data "watched" list"""
+    user_data["watched"].append(
+        create_movie(movie["title"], movie["genre"], movie["rating"])
+    )
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    """append movie dictionairy to user_data "watchlist" list"""
+    user_data["watchlist"].append(
+        create_movie(movie["title"], movie["genre"], movie["rating"])
+    )
+    return user_data
+
+
+def watch_movie(user_data, title):
+    """Move a movie from user_data "watchlist" list to "watched" list"""
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data

```

Comment:
[warning] Here we are iterating over the user's watchlist, and also modifying it (`remove`). We should avoid modifying a list while iterating over it to avoid problems with the iteration. _If_ we stop iterating as soon as we make a change, it would still be safe (though anyone else reading the code would need to convince themselves of this). However, depending on our data restrictions, this code could end up modifying the list multiple times if there is more than one movie with the same title.

To ensure that only one change is made, we could break out as soon as we find the movie and move it. Again, this is safe from an iteration perspective, but may require other team members to spend time convincing themselves of this. We might instead decide to think of this as 2 steps. 1. Find the movie to move by iterating through the list, and 2. actually move the move (outside the list). This might be more clear that there are no negative modifying while iterating shenanigans!

```python
        for movie in user_data["watchlist"]:
            if movie["title"] == title:
                user_data["watched"].append(movie)
                user_data["watchlist"].remove(movie)
                break

        return user_data
```